### DISCUSSION TITLE
Generate images in colab without WebUI

### DISCUSSION THREAD

**Discussion by harsh1kumar** [lllyasviel/Fooocus#1610] [Q&A]
Hello Folks,

I have generated images using WebUI started using colab. However, I want to be able to generate images programatically. I mean my prompt will be generated through some python code.

How can I get predicted images by writing a python code in colab? Is there any guide for this? All the instructions seem to call `python entry_with_update.py`, which launches the webUI.

I tried using gradio_client API for `fn_index: 32`, but that generates errors which I have not been able to fix. Any tutorial/documentation which I can read for this? (I can share my code and error if needed.)

**Comment by koriekhov** [comment]
Work with gradio API is quite uncomfortable.
You may try to use some Fooocus forks, like Fooocus-API https://github.com/konieshadow/Fooocus-API

**Reply by harsh1kumar** [reply]
Ahh okay, let me check that out. Thanks @koriekhov
